File: src/bin/day-01.py
from pathlib import Path
from typing import Tuple

inputs = Path("inputs/day-01.txt").read_text()


def solution() -> Tuple[int, int]:
    calories: list[int] = list(
        map(
            lambda elf: sum(map(lambda i: int(i.strip()), elf.splitlines())),
            inputs.split("\n\n"),
        )
    )

    calories.sort(reverse=True)

    return calories[0], sum(calories[0:3])


part1, part2 = solution()
print(f"Part 1: {part1} / Part 2: {part2}")

File: 2022/src/bin/day-01.rs
const INPUT: &str = include_str!("../../inputs/day-01.txt");

fn solution() -> (i32, i32) {
    let mut calories = INPUT
        .split("\n\n") // Split by elf
        .map(|elf| {
            elf
                .lines()
                .map(|line| line.trim().parse::<i32>().expect("Expected a number"))
                .sum()
        })
        .collect::<Vec<i32>>();
    
    calories.sort();
    calories.reverse();
    
    (calories[0], calories[0..3].into_iter().sum())
}

fn main() {
    let (part1, part2) = solution();
    println!("Part 1: {part1} / Part 2: {part2}");
}
